src/OmniSharp.Roslyn.CSharp/Services/Refactoring/V2/BaseCodeActionService.cs

```diff
@@ -149,58 +142,36 @@ private async Task AppendFixesAsync(Document document, TextSpan span, IEnumerabl
             }
         }
 
-        private List<CodeFixProvider> GetSortedCodeFixProviders()
+        private List<CodeFixProvider> GetSortedCodeFixProviders(Document document)
         {
-            var providerList = this.Providers.SelectMany(provider => provider.CodeFixProviders);
+            var providerList =
+                this.Providers.SelectMany(provider => provider.CodeFixProviders)
+                    .Concat(codeFixesForProject.GetAllCodeFixesForProject(document.Project.Id.ToString()));
+
             return ExtensionOrderer.GetOrderedOrUnorderedList<CodeFixProvider, ExportCodeFixProviderAttribute>(providerList, attribute => attribute.Name).ToList();
```

Comment:
I dont fully understand what all this sorting does. but probably good to profile to see whether it is doing things efficiently. if expected dataset is most likely always small then, probably doesn't care.

Reply:
will it be called every time user move caret?

Reply:
I think this is only called when available code actions for current cursor location is requested. In vscode this happens when ctrl+. is pressed and menu of actions is shown